homeassistant/components/shelly/event.py

```diff
@@ -17,25 +19,37 @@
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
 from homeassistant.helpers.update_coordinator import CoordinatorEntity
 
-from .const import RPC_INPUTS_EVENTS_TYPES
-from .coordinator import ShellyRpcCoordinator, get_entry_data
+from .const import (
+    BASIC_INPUTS_EVENTS_TYPES,
+    RPC_INPUTS_EVENTS_TYPES,
+    SHIX3_1_INPUTS_EVENTS_TYPES,
+)
+from .coordinator import ShellyBlockCoordinator, ShellyRpcCoordinator, get_entry_data
 from .utils import (
     async_remove_shelly_entity,
     get_device_entry_gen,
     get_rpc_input_name,
     get_rpc_key_instances,
+    is_block_momentary_input,
     is_rpc_momentary_input,
 )
 
 
 @dataclass
-class ShellyEventDescription(EventEntityDescription):
+class ShellyRpcEventDescription(EventEntityDescription):
     """Class to describe Shelly event."""
 
     removal_condition: Callable[[dict, dict, str], bool] | None = None
 
 
-RPC_EVENT: Final = ShellyEventDescription(
+@dataclass
+class ShellyBlockEventDescription(EventEntityDescription):
```

Comment:
In all of the other platforms we keep the Block (Gen1) related definitions before Rpc (Gen2) definitions, this keeps easier tracking Gen1/Gen2 and is also sorted alphabetically.

Reply:
Good point, thanks.